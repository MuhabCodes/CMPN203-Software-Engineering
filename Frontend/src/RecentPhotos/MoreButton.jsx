import React from 'react';

const MoreButton = () => (
  <button className="dropdown-toggle" type="button" data-testid="more-test" id="exp-dropdown-menu-button1" data-toggle="dropdown" aria-expanded="false">
    More
  </button>
);
export default MoreButton;
