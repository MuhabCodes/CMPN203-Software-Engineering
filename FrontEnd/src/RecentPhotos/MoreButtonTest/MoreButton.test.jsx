import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import MoreButton from '../MoreButton';
import '@testing-library/jest-dom/extend-expect';

it('it renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MoreButton />, div);
});

it('renders the button correctly', () => {
  const { getByTestId } = render(<MoreButton />);
  expect(getByTestId('more-test')).toBeTruthy();
});

it('matches a snapshot', () => {
  const tree = renderer.create(<MoreButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
