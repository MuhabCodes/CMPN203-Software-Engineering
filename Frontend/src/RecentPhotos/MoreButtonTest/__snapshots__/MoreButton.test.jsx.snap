// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<button
  aria-expanded="false"
  className="dropdown-toggle"
  data-testid="more-test"
  data-toggle="dropdown"
  id="exp-dropdown-menu-button1"
  type="button"
>
  More
</button>
`;
